### PR TITLE
distutils-r1.eclass: DISTUTILS_USE_PEP517=no mode

### PR DESCRIPTION
**Pull request by mgorny**
Introduce a new DISTUTILS_USE_PEP517 value "no" that stands for
"no build system".  This is primarily meant to replace the legacy
distutils-r1 logic used for bootstrapping baseline PEP 517 packages.
At the same time, it provides a convenient replacement for some
of the uses of python-r1.

In this mode, the eclass does not add PEP517-specific dependencies
and does not export default python_compile() and python_install()
implementations.  However, it does set dependencies, REQUIRED_USE
and enables sub-phase usage (with respect to DISTUTILS_OPTIONAL).
It also permits using distutils_enable_{sphinx,tests}.

### CODE REVIEW

**Review by arthurzam — APPROVED** [review]

File: app-admin/gnome-abrt/gnome-abrt-1.4.1-r1.ebuild [hunk 1]
```diff
@@ -0,0 +1,58 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+DISTUTILS_USE_PEP517=no
```
Comment (ionenwks) [inline app-admin/gnome-abrt/gnome-abrt-1.4.1-r1.ebuild:6]: gnome-abrt-1.4.2 is in the tree, probably need to update this one

**Review by thesamesam — APPROVED** [review]
I couldn't tell you why but I felt a bit on the fence about this, but I can't give any actual good reason for it, the change is sensible, and I can't suggest anything better. So let's go with it.

(Your potential changes mentioned in -python earlier notwithstanding, but wanted to say I'm fine w/ this version.)

EDIT: Oh, I realised my objection. I'm worried that this is making it less clear when to use python-r1 vs distutils-r1.

**Comment by mgorny** [comment]
I think that's because "no" is going to mean something different from unset for the time being — but I honestly can't think of a better long-term value.

**Review by arthurzam — APPROVED** [review]

File: app-admin/gnome-abrt/gnome-abrt-1.4.1-r1.ebuild (hunk 1 above)
Comment (mgorny) [reply app-admin/gnome-abrt/gnome-abrt-1.4.1-r1.ebuild:6]: It's just an example, I'm not going to merge it.

**Comment by mgorny** [comment]
> EDIT: Oh, I realised my objection. I'm worried that this is making it less clear when to use python-r1 vs distutils-r1.

I think the "rule of thumb" still applies, i.e. whatever is easier. Ideally we wouldn't have to use python-r1 directly at all.

File: eclass/distutils-r1.eclass [hunk 2]
```diff
@@ -1854,6 +1915,10 @@ _distutils-r1_check_namespace_pth() {
 		ewarn "the ebuild accordingly:"
 		ewarn
 		ewarn "  https://projects.gentoo.org/python/guide/concept.html#namespace-packages"
+
+		if ! has "${EAPI}" 5 6 7 8; then
```
Comment (arthurzam) [inline eclass/distutils-r1.eclass:1919]: I think `EAPI=5` is already banned for this eclass?

**Review by thesamesam — APPROVED** [review]

File: eclass/python-utils-r1.eclass [hunk 3]
```diff
@@ -854,13 +869,24 @@ python_domodule() {
...
+			doins -r "${@}" || return ${?}
+		)
+		python_optimize "${ED%/}/${d}"
+	elif [[ -n ${BUILD_DIR} ]]; then
+		local dest=${BUILD_DIR}/install${EPREFIX}/${d}
+		mkdir -p "${dest}" || die
+		cp -pR "${@}" "${dest}/" || die
+		(
+			cd "${dest}" &&
+			chmod -R a+rX -R "${@##*/}"
```
Comment (ionenwks) [inline eclass/python-utils-r1.eclass:885]: has two -R

File: eclass/distutils-r1.eclass (hunk 2 above)
Comment (mgorny) [reply eclass/distutils-r1.eclass:1922]: You're right.

File: eclass/python-utils-r1.eclass (hunk 3 above)
Comment (mgorny) [reply eclass/python-utils-r1.eclass:885]: Better safe than sorry!

**Review by arthurzam — APPROVED** [review]

**Comment by mgorny** [comment]
After sending to the ml on Saturday, I'd like to merge it today. Could you please take one last look?

**Review by ionenwks — APPROVED** [review]

**Review by arthurzam — APPROVED** [review]

**Review by thesamesam — APPROVED** [review]

**Pull request closed without merging**